frontend/src/modules/editor/themeEditor/views/editorThemingView.js

```diff
@@ -0,0 +1,576 @@
+// LICENCE https://github.com/adaptlearning/adapt_authoring/blob/master/LICENSE
+define(function(require) {
+  var Backbone = require('backbone');
+  var EditorOriginView = require('../../global/views/editorOriginView');
+  var Helpers = require('core/helpers');
+  var Origin = require('core/origin');
+  var ThemeCollection = require('../collections/editorThemeCollection');
+  var PresetCollection = require('../collections/editorPresetCollection.js');
+  var PresetEditView = require('./editorPresetEditView.js');
+  var PresetModel = require('../models/editorPresetModel.js');
+
+  var ThemingView = EditorOriginView.extend({
+    tagName: 'div',
+    className: 'theming',
+
+    settings: {
+      presetSelection: null
+    },
+
+    events: {
+      'change .theme select': 'onThemeChanged',
+      'change .preset select': 'onPresetChanged',
+      'change .form-container form': 'updateRestorePresetButton',
```

Comment:
I'm fine with putting this back into the onFieldChanged handler, please could you explain how it's DRYer though? I just figured it made more sense to have the event call the function directly rather than call a function whose only purpose is to call another function?

Reply:
Sure. I envisioned the UI to be tweaked somehow to make it obvious we've customised our theme/preset, e.g.:
![modified](https://user-images.githubusercontent.com/922987/56348336-1ecde000-61be-11e9-8850-5827691db32e.png)
I nearly requested this in a review but I didn't want to burden the workload any more at this point! With the current code, any further UI changes would require repeating the event handler:
```js
'change .form-container form': 'updateRestorePresetButton',
'change .form-container form': 'updateSelectText'
```
The other undesired side effect of not having a dedicated event handler, as you've already noticed, is that we don't have control over the arguments. The logic had to be changed [here](https://github.com/adaptlearning/adapt_authoring/pull/2209/commits/97179a923c5cda712cc5e9d3dcdd1e94db4e7cd4#diff-18eb4043de18e1156fb7293f6fa554e1R453) because the `shouldShow` parameter is now sometimes the change event.

In an ideal world, there would be no comparisons being done in the UI functions – instead we'd derive `shouldShow`/`isThemeModified` in the event handler or even in its own function so it's reusable, and then pass a toggle to one/more UI setters.

Hope I'm making sense!

Reply:
Ok fair point, I will put it back to how it was. Thanks for explaining.

I would have created an event handler function instead of having the same event calling multiple functions for sure but I see your point about the logic.

I like the idea of tweaking the UI in this way, happy to raise an issue for it and work on it (perhaps as part of a bug patch) but I agree we shouldn't be adding more to this PR/release at this point.